MediaBrowser.Common/Plugins/LocalPlugin.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace MediaBrowser.Common.Plugins
+{
+    /// <summary>
+    /// Local plugin struct.
+    /// </summary>
+    public readonly struct LocalPlugin : IEquatable<LocalPlugin>
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalPlugin"/> struct.
+        /// </summary>
+        /// <param name="id">The plugin id.</param>
+        /// <param name="name">The plugin name.</param>
+        /// <param name="version">The plugin version.</param>
+        /// <param name="path">The plugin path.</param>
+        public LocalPlugin(Guid id, string name, Version version, string path)
+        {
+            Id = id;
+            Name = name;
+            Version = version;
+            Path = path;
+            DllFiles = new List<string>();
+        }
+
+        /// <summary>
+        /// Gets the plugin id.
+        /// </summary>
+        public Guid Id { get; }
+
+        /// <summary>
+        /// Gets the plugin name.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// Gets the plugin version.
+        /// </summary>
+        public Version Version { get; }
+
+        /// <summary>
+        /// Gets the plugin path.
+        /// </summary>
+        public string Path { get; }
+
+        /// <summary>
+        /// Gets the list of dll files for this plugin.
+        /// </summary>
+        public List<string> DllFiles { get; }
+
+        /// <summary>
+        /// == operator.
+        /// </summary>
+        /// <param name="left">Left item.</param>
+        /// <param name="right">Right item.</param>
+        /// <returns>Comparison result.</returns>
+        public static bool operator ==(LocalPlugin left, LocalPlugin right)
+        {
+            return left.Equals(right);
+        }
+
+        /// <summary>
+        /// != operator.
+        /// </summary>
+        /// <param name="left">Left item.</param>
+        /// <param name="right">Right item.</param>
+        /// <returns>Comparison result.</returns>
+        public static bool operator !=(LocalPlugin left, LocalPlugin right)
+        {
+            return !(left == right);
```

Comment:
Although this works, wouldn't this be optimum?
```suggestion
            return !left.Equals(right);
```